orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/AbstractClusterWideClouddriverTask.groovy

```diff
@@ -201,6 +176,54 @@ abstract class AbstractClusterWideClouddriverTask extends AbstractCloudProviderA
     return filterParentDeploys(stage, account, location, serverGroups)
   }
 
+  static List<TargetServerGroup> parentDeployedServerGroups(Stage stage,
+                                                            String account,
+                                                            Location location,
+                                                            List<TargetServerGroup> clusterServerGroups) {
+    //if we are a synthetic stage child of a deploy, don't operate on what we just deployed
+    final Set<String> deployStageTypes = [
+      DeployStage.PIPELINE_CONFIG_TYPE,
+      CopyLastAsgStage.PIPELINE_CONFIG_TYPE,
+      CloneServerGroupStage.PIPELINE_CONFIG_TYPE,
+      CreateServerGroupStage.PIPELINE_CONFIG_TYPE,
+    ]
+    List<TargetServerGroup> deployedServerGroups = []
+
+    stage.ancestors({ Stage ancestorStage, StageBuilder stageBuilder ->
+      // Stage type is the context.type value when the stage is running as a child stage of a parallel deploy, or
+      // the stage.type attribute when it is running directly as part of an Orchestration or Pipeline
+      (deployStageTypes.contains(ancestorStage.type) || deployStageTypes.contains(ancestorStage.context.type)) && ancestorStage.context.'deploy.account.name' == account
+    })*.stage.each { Stage parentDeployStage ->
+      Map<String, String> dsgs = (parentDeployStage.context.'deploy.server.groups' ?: [:]) as Map
+      switch (location.type) {
+        case Location.Type.ZONE:
+          deployedServerGroups.addAll(clusterServerGroups.findAll {
+            it.zones.contains(location.value) && dsgs[it.region].contains(it.name)
+          })
+          break;
+        case Location.Type.REGION:
+          deployedServerGroups.addAll(clusterServerGroups.findAll {
+            it.region == location.value && dsgs[location.value].contains(it.name)
+          })
+          break;
+      }
+    }
+
+    return deployedServerGroups
+  }
+
+  static List<TargetServerGroup> filterParentAndNewerThanParentDeploys(List<TargetServerGroup> parentDeployedServerGroups,
+                                                                       List<TargetServerGroup> clusterServerGroups) {
+    def recentlyDeployedServerGroups = []
+    if (parentDeployedServerGroups) {
+      Long minCreatedTime = (parentDeployedServerGroups*.createdTime as Collection<Long>).min()
```

Comment:
In the event of their being multiple upstream deployed server groups, I opt'd to take the earliest creation time and then skipping anything created after it.

This seemed reasonable but I'm open to opinions.